### PR TITLE
Added OpenID Json Web Key Sets for verification via the @auth directive

### PR DESCRIPTION
**Pull request by dvanmali**
# Description

According to OpenID configuration, verification of JWTs should be validated against the authentication server's JWKS to ensure that the server signed it. Instead of interpreting the JWT via a secret or PEM file (which if maliciously stolen, could rewrite JWTs), we verify via the signature represented by the token's kid at the endpoint `https://YOUR_DOMAIN/.well-known/jwks.json`. 

These changes are currently non-breaking changes.

# Issue

#252 

Passing a decoded token only partially solves this verification issue. We'd rather perform verification via the GraphQL @auth directive so checks occur at each request and are done **without secrets**.

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-564.surge.sh

File: packages/graphql/src/auth/get-jwt.ts [hunk 1]
```diff
@@ -68,8 +70,33 @@ function getJWT(context: Context): any {
...
             result = jsonwebtoken.decode(token);
-        } else {
-            debug("Verifying JWT");
+        } else if (jwtConfig.jwkEndpoint) {
+            debug("Verifying JWT using OpenID Public Key Endpoint");
+
+            // Create a JWK Client with a rate limit that
+            // limits the number of calls to our JWK endpoint
+            client = new JwksClient({
+                jwksUri: jwtConfig.jwkEndpoint || "",
```
Comment (oskarhane) [inline packages/graphql/src/auth/get-jwt.ts:79]: Do we really need a default value here? At this point the `jwtConfig.jwkEndpoint` should always be set, right?

**Review by oskarhane — COMMENTED** [review]
Thank you for this PR @dvanmali, really appreciate it.
Looks good in general, just a small comment.

Also, could you come up with a way to test this (that it takes the code path when the config value is set) so we don't introduce regressions in the future?

**Review by oskarhane — APPROVED** [review]
LGTM, thanks a lot @dvanmali

**Pull request merged**